OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -766,16 +766,52 @@ void TracerThread::ProcessExitEvent(const perf_event_header& header,
 
 void TracerThread::ProcessMmapEvent(const perf_event_header& header,
                                     PerfEventRingBuffer* ring_buffer) {
-  pid_t pid = ReadMmapRecordPid(ring_buffer);
+  // Mmap records have the following layout:
+  // struct {
+  //   struct perf_event_header header;
+  //   u32    pid, tid;
+  //   u64    addr;
+  //   u64    len;
+  //   u64    pgoff;
+  //   char   filename[];
+  //   struct sample_id sample_id; /* if sample_id_all */
+  // };
+  // Because of filename, the layout is not fixed.
+
+  perf_event_sample_id_tid_time_streamid_cpu sample_id;
+  ReadPerfSampleIdAll(ring_buffer, header, &sample_id);
+
+  perf_event_mmap_up_to_filename mmap_event;
+  ring_buffer->ReadValueAtOffset(&mmap_event, 0);
+
+  //  ring_buffer->ConsumeRecord()
+  // read filename
+  size_t filename_offset = sizeof(perf_event_mmap_up_to_filename);
+  // strictly > because filename is null-terminated string
+  CHECK(header.size > (filename_offset + sizeof(perf_event_sample_id_tid_time_streamid_cpu)));
+  size_t filename_size =
+      header.size - filename_offset - sizeof(perf_event_sample_id_tid_time_streamid_cpu);
+  std::vector<char> filename_vector(filename_size);
+  ring_buffer->ReadRawAtOffset(&filename_vector[0], filename_offset, filename_size);
+  std::string filename(&filename_vector[0], filename_size);
+
   ring_buffer->SkipRecord(header);
 
-  if (pid != target_pid_) {
+  if (static_cast<pid_t>(sample_id.pid) != target_pid_) {
+    return;
+  }
+
+  if (sample_id.time < effective_capture_start_timestamp_ns_) {
     return;
   }
 
-  // There was a call to mmap with PROT_EXEC, hence refresh the maps. This should happen rarely.
+  // There was a call to mmap with PROT_EXEC, hence refresh the maps.
+  // This should happen rarely.
+  // Note that we do not use filename from the record itself instead we read all maps
+  // from /proc/pid/maps. Consider moving this to MMAP2 event which has more information (like
+  // flags)
```

Comment:
Isn't this comment wrong? Isn't this moving away from reading `/proc/pid/maps`? Isn't `filename` actually used?

Reply:
yep - updated this as well.